src/renderer/domains/network/account-sync/resource.ts

```diff
@@ -105,6 +133,9 @@ export const proxyAccountsProvider: AccountProvider<SyncedProxyAccount> = {
           });
           if (nullable(proxyFromIndexer)) continue;
 
+          // Check if this account is a pure proxy using the PureProxy table data
+          const isPureProxy = pureProxyLookup.has(`${accountId}:${chainId}`);
```

Comment:
maybe we can keep the old check too
```suggestion
          const isPureProxy = proxyFromIndexer.isPureProxy || pureProxyLookup.has(`${accountId}:${chainId}`);
```